### PR TITLE
Env, add environment component

### PR DESCRIPTION
**Pull request by mruz**

### CODE REVIEW

File: ice/env.zep [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+     * Env constructor.
+     *
+     * @param string data Path to the env file
+     */
+    public function __construct(var data = null)
+    {
+        // No support for passing variables by reference yet zephir/issues/203
+        //let this->data = &_ENV;
+
+        parent::__construct(data);
```
Comment (Yahasana) [inline ice/env.zep:28]: `php` config in favor of `ini`, which has to load config from an ini file.
```php
class Env extends Config
{
    public function __construct(var data = null)
    {
        parent::__construct(data);

        let _ENV = array_merge(_ENV, this->data);
    }
    // more methods need to overide
    public function getData()
    {
        return _ENV;
    }
    public function keys() -> array
    {
        return array_keys(_ENV);
    }
    public function count() -> int
    {
        return count(_ENV);
    }
}

$config = new Env(['mode' => 'debug']);
// or
$config = new Env((new Ini('env.ini'))->getData());

File: ice/env.zep [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        return defaultValue;
+    }
+
+    /**
+     * Assigns a value to the specified environment key.
+     * @return object
+     */
+    public function set(string key, var value) -> <self>
+    {
+        parent::set(key, value);
```
Comment (Yahasana) [inline ice/env.zep]: `parent::set(key, value);` is no needed

File: ice/env.zep [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        return this;
+    }
+
+    /**
+     * Remove environment key.
+     * @return object
+     */
+    public function remove(string key) -> <self>
+    {
+        parent::erase(key);
```
Comment (Yahasana) [inline ice/env.zep]: `parent::erase(key);` is no needed. all get/set/del operations base on _ENV are enough

File: ice/env.zep (hunk 2 above)
Comment (Yahasana) [inline ice/env.zep]: no need to `parent::set(key, value)`. all get/set/erase operations are base on `_ENV`

File: ice/env.zep (hunk 3 above)
Comment (Yahasana) [inline ice/env.zep]: no need to `parent::erase(key)`. all get/set/erase operations are base on `_ENV`

File: ice/env.zep (hunk 1 above)
Comment (mruz) [reply ice/env.zep:28]: But on `.env` file you can put your environment variables as key-value pairs
```ini
DB_NAME=database name
DB_PASSWORD=@$fq42$d2r2
DB_HOST="localhost"
```
syntax is more similar to ini, so I used that adapter.

File: ice/env.zep (hunk 2 above)
Comment (mruz) [reply ice/env.zep]: But then a key will be missing on this->data.

Comment (Yahasana) [reply ice/env.zep]: get / set do not operate on `this->data`.  copy config data to `this->data` is meaningless.
```php
$env1 = new Env;
$env2 = new Env;
$env1->set("foo", "bar");
$env2->set("xxx", "zzz");

// expected both env1 & env2 data contain foo & xxx. but they do not. 
// both share the same _ENV but different this->data
print_r($env1->data);
print_r($env2->data);
```

File: ice/env.zep (hunk 1 above)
Comment (Yahasana) [reply ice/env.zep:28]: `Env` class is helper to maintain superglobal variable $_ENV, but not target to `env.ini` file.
 
for this case `$config = new Env((new Ini('env.ini'))->getData())`

Comment (mruz) [reply ice/env.zep:28]: Right, but not env.ini, but .env, like https://github.com/vlucas/phpdotenv

**Comment by Yahasana** [comment]
why don't place `env.zep` into `config` folder?

**Comment by mruz** [comment]
It's a helper for global variable like Session, so I put it in the root folder. Do you think it should go to Config folder?

**Comment by Yahasana** [comment]
I don't think this helper is worth adding to Ice but hiding it in config folder is ok
> Please note that writing to $_ENV does not actually set an environment variable, i.e. the variable will not propagate to any child processes you launch (except forked script processes, in which case it's just a variable in the script's memory). To set real environment variables, you must use putenv().
>
> Basically, setting a variable in $_ENV does not have any meaning besides setting or overriding a script-wide global variable. Thus, one should never modify $_ENV except for testing purposes (and then be careful to use putenv() too, if appropriate).
>
> PHP  will not trigger any kind of error or notice when writing to $_ENV.

**Pull request merged**